### PR TITLE
Added Alert for ControlPanel and ChartContainer

### PR DESCRIPTION
**Pull request by vera-liu**
Done:
 - Add alert for Control Panel when fetch_datasource_metadata failes
 - Add alert for Chart Container when update_explore query fails

Control Panel alerts:
screen shot 2016-11-17 at 3 30 40 pm

Chart Container alerts:
screen shot 2016-11-17 at 3 29 54 pm

needs-review @ascott @bkyryliuk @mistercrunch

### CODE REVIEW

**Comment by ascott** [comment]
this looks great! nice work! 

i wonder if we should use the yellow alert here rather than blue? i'm not a big fan of the red/danger alert, but i think the yellow works well for these type of fetching errors/user feedback.

screenshot 2016-11-17 12 53 03

**Comment by vera-liu** [comment]
Changed color :) @ascott

**Comment by ascott** [comment]
LGTM once tests are passing :ship:

File: superset/views.py [hunk 1]
```diff
@@ -1301,9 +1301,14 @@ def update_explore(self, datasource_type, datasource_id):
                 datasource_id=datasource_id,
                 args=form_data)
         except Exception as e:
-            flash('{}'.format(e), "alert")
-            return redirect(error_redirect)
-        return viz_obj.get_json()
+            logging.exception(e)
```
Comment (mistercrunch) [inline superset/views.py]: `error_redirect` (line 1297) isn't used anymore and should be removed

File: superset/views.py [hunk 2]
```diff
@@ -1301,9 +1301,14 @@ def update_explore(self, datasource_type, datasource_id):
...
+            return json_error_response('{}'.format(e))
+        try:
+            payload = viz_obj.get_json()
+        except Exception as e:
+            logging.exception(e)
+            return json_error_response(utils.error_msg_from_exception(e))
+        return payload
```
Comment (mistercrunch) [inline superset/views.py]: payload may be out of scope here, maybe the try block should `return viz_obj.get_json()`

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx [hunk 3]
```diff
@@ -183,6 +187,16 @@ class ChartContainer extends React.Component {
             </div>
           }
         >
+          {this.props.alert &&
+            <Alert bsStyle="warning">
+            {this.props.alert}
```
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: indent

**Review by mistercrunch — CHANGES_REQUESTED** [review]

**Pull request merged**